### PR TITLE
Add device id to hardware info

### PR DESCRIPTION
**Pull request by spylogsster**
Native side of https://github.com/brave/brave-browser/issues/18165 

- Remove indexes from device account names
- Return device id in account info payload to group devices by id
- Moved device id calculation to javascript side when we unlock device to constantly calculate it on zero address because user can select another addresses to import

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- import accounts from device
- check that names do not have indexes
- check that if import N-th accounts in different times from device, they will be under same device section in Preferences

### CODE REVIEW

File: components/brave_wallet_ui/common/ledgerjs/eth_ledger_bridge_keyring.ts [hunk 1]
```diff
@@ -63,6 +67,16 @@ export default class LedgerBridgeKeyring extends EventEmitter {
...
   /* PRIVATE METHODS */
+  _getDeviceId = async () => {
+    const zeroPath = this._getPathForIndex(0, LedgerDerivationPaths.LedgerLive)
+    const address = await this._getAddress(zeroPath)
+    const msgBuffer = new TextEncoder().encode(address)
+    const hashBuffer = await crypto.subtle.digest('SHA-256', msgBuffer)
+    const hashArray = Array.from(new Uint8Array(hashBuffer))
+    const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('')
+    return hashHex
+  }
```
[the remark this answers is not in the document]
Comment (darkdh) [reply components/brave_wallet_ui/common/ledgerjs/eth_ledger_bridge_keyring.ts:78]: the device id used here is unique per seed, if users use multiple hardware devices with same seed, they will be treated as same device. WebHID doesn't give us unique ids for different devices with same model.

[the remark this answers is not in the document]
Comment (darkdh) [reply components/brave_wallet_ui/common/ledgerjs/eth_ledger_bridge_keyring.ts:78]: Also what we need from these different devices with same seed is no different. address and private key used to sign transaction is the same.

[the remark this answers is not in the document]
Comment (onyb) [reply components/brave_wallet_ui/common/ledgerjs/eth_ledger_bridge_keyring.ts:78]: Thanks for the clarification. There's the "key fingerprint" from BIP32 spec that we could use, but hash form zero indexed address is not much different. Resolving this conversation.

**Review by onyb — APPROVED** [review]
Looks good for changes in `eth_ledger_bridge_keyring.ts`.

**Pull request merged**